### DISCUSSION TITLE
An idea of introducing forward looking sonar model for underwater environment

### DISCUSSION THREAD

**Discussion by AaaGss** [cyberbotics/webots#6486] [Ideas]
Hello Development Team, 

I am a researcher who focuses on underwater applications. Recently, I started using the Webots software and encountered some issues. It appears that the software does not include simulation models for sonar or forward-looking sonar, despite their widespread use in underwater environments, similar to how cameras are used on land. 

I would like to request that the development team consider integrating forward-looking sonar and sonar sensors into the software. Doing so would greatly improve the software's capabilities and make it more useful for underwater research. 

Thank you for your attention to this matter.

**Comment by ShuffleWire** [comment]
Could the Radar node  (https://cyberbotics.com/doc/reference/radar) could be used instead ?

Additionally to make a front facing one you just have to use the horizontalFieldOfView / verticalFieldOfView properties.

**Comment by AaaGss** [comment]
Although both radar and sonar use polar coordinates, this is feasible for target location. However, sonar can also map the underwater environment and generate sonar images. For image generation, the radar module of the software is unable to generate images. This is a great obstacle for the research of underwater sonar image processing. That's why I recommend that the software integrate sonar sensors.




946175224
***@***.***



&nbsp;





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cyberbotics/webots"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月19日(星期二) 下午4:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cyberbotics/webots] An idea of introducing forward looking sonar model for underwater environment (Discussion #6486)





 
Could the Radar node  (https://cyberbotics.com/doc/reference/radar) could be used instead ?
 
Additionally to make a front facing one you just have to use the horizontalFieldOfView / verticalFieldOfView properties.
